src/foam/java/PropertyInfo.js

```diff
@@ -155,111 +148,119 @@ foam.CLASS({
             body: '((' + this.sourceCls.name + ') o).' + this.clearName + '();'
           },
           {
+            name: 'isSet',
+            visibility: 'public',
+            type: 'boolean',
+            args: [{ name: 'o', type: 'Object' }],
+            body: `return ((${this.sourceCls.name}) o).${this.propName}IsSet_;`
+          }
+        ];
+        var primitiveType = ['boolean', 'long', 'byte', 'double','float','short','int'];
+        
+        if ( this.propType  == 'java.util.Date' || ! ( primitiveType.includes(this.propType) || this.propType == 'Object' || this.propType == 'String') ){
+          m.push({
             name: 'cast',
             type: this.propType,
             visibility: 'public',
             args: [{ name: 'o', type: 'Object' }],
             body: 'return ' + ( this.propType == "Object" ? 'o;' : '( ' + this.propType + ') o;')
-          },
-          {
-            name: 'compare',
-            type: 'int',
-            visibility: 'public',
-            args: [{ name: 'o1', type: 'Object' }, { name: 'o2', type: 'Object' }],
-            body: this.compare,
-          },
-          {
-            name: 'comparePropertyToObject',
-            type: 'int',
+          });
+        }
+        
+        if ( this.propType  == 'java.util.Date' || this.propType == 'String' ||  ! ( primitiveType.includes(this.propType)|| this.propType == 'Object' || this.extends == 'foam.core.AbstractFObjectPropertyInfo' || this.extends == 'foam.core.AbstractFObjectArrayPropertyInfo') ){
```

Comment:
Can you split into multiple lines - it will make future diffs easier to see.  And the few that follow, thanks.
a ||
b ||
c...